core/src/apps/management/recovery_device/word_validity.py

```diff
@@ -0,0 +1,103 @@
+from micropython import const
+
+import storage.recovery
+from trezor.messages import BackupType
+
+from apps.management.recovery_device import recover
+
+if False:
+    from typing import List, Optional
+    from trezor.messages.ResetDevice import EnumTypeBackupType
+
+OK = const(0)
+NOK_IDENTIFIER_MISMATCH = const(1)
+NOK_ALREADY_ADDED = const(2)
+NOK_THRESHOLD_REACHED = const(3)
+
+
+def check(
+    current_index: int,
```

Comment:
this argument is superfluous and should always derive from length of `previous_words` unless i'm mistaken?

the "correct" signature for this method would be something like `check(backup_type: BackupType, partial_mnemonic: List[str])`. There isn't even a reason to keep `current_word` separate, we can do `words.append(word)` at the calling site and it doesn't seem to affect anything?

Reply:
IMHO yes. This was written by @ciny originally, maybe he has something to add?

Reply:
I have rewritten it and it is indeed cleaner. Check it out in https://github.com/trezor/trezor-firmware/pull/774/commits/f4e11a9176af83fcdde5b1edc0d584585ca1bfe9